compiler/daml-lf-evaluator/src/DA/Daml/LF/Evaluator/Exp.hs

```diff
@@ -0,0 +1,42 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.Evaluator.Exp
+  ( Exp(..), Alt(..), Var, Prog(..), Defs, DefKey(..), FieldName, TyVar, Type,
+  ) where
+
+import qualified DA.Daml.LF.Ast as LF
+import DA.Daml.LF.Evaluator.Value(Value,FieldName,Tag)
+import Data.Map.Strict (Map)
+
+data Exp
```

Comment:
What advantage does this give you over using LF.Expr directly?

Reply:
So the evaluator code was written about 5 months ago.
The idea was to write an small evaluator, as my first steps in learning about DAML-LF.

To keep thing simple. I introduced a simplified AST, and wrote the evaluator in two phases

    LF --(Simp)--> SIMP-AST --(Eval)--> Value

It was then I began to explore normilization, and experimented with:

    LF --(Simp)--> SIMP-AST --(Old-Norm)--> SIMP-AST --(Eval)--> Value

The SIMP-AST was a nice place to pretty-print the essence of the program being normalized, without getting lost in all the details of full LF. And with some instrumentation on the evaluator I got some idea of how much reduction could be saved.

Now, normalizing in SIMP-AST is not very useful; we want to normalize directly on the LF.
This is what  have been playing over the last few days.

So now we can:

    LF --(Norm)--> LF

As before I can continue to use my own simple evaluator if desired.

    LF --(Norm)--> LF  ... --(Simp)--> SIMP-AST --(Eval)--> Value

But better still, we can write the optimized DAR to file, and try it for real.

    daml build --run-optimizer

The new LF-normalization code is here:
    compiler/daml-lf-optimize/src/DA/Daml/LF/Optimize.hs

The code in `compiler/daml-lf-evaluator` can mostly be considered throw away. Or only for testing.